### PR TITLE
Removing deprecated version-number option

### PR DESCRIPTION
**Pull request by lucasncabral**
This change affects the `ml-git commit` and `ml-git create` commands. The user must use the `--version` option instead of old deprecated option `--version-number`.

```
Usage: ml-git dataset commit [OPTIONS] ML_ENTITY_NAME

  Commit dataset change set of ML_ENTITY_NAME locally to this ml-git
  repository.

Options:
  --tag TEXT               Ml-git tag to identify a specific version of a ML
                           entity.
  --version INTEGER RANGE  Set the number of artifact version.
  -m, --message TEXT       Use the provided <msg> as the commit message.
  --fsck TEXT              Run fsck after command execution.
  --verbose                Debug mode
```

### CODE REVIEW

File: docs/mlgit_commands.md [hunk 1]
```diff
@@ -154,9 +154,7 @@ Options:
                                   version.
   --tag TEXT                      Ml-git tag to identify a specific version of
                                   a ML entity.
-  --version-number, --version INTEGER RANGE
-                                  Set the number of artifact version.
-                                  [DEPRECATED:--version-number]
+  --version INTEGER RANGE         Set the number of artifact version.
```
Comment (andersonpablosilva) [inline docs/mlgit_commands.md:157]: Why are we specifying an INTEGER RANGE? Maybe you should use just INTEGER.

File: docs/mlgit_commands.md [hunk 2]
```diff
@@ -196,9 +194,7 @@ Options:
   --store-type, --storage-type [s3h|azureblobh|gdriveh]
                                   Data storage type [default: s3h].
                                   [DEPRECATED:--store-type]
-  --version-number, --version INTEGER RANGE
-                                  Number of artifact version.
-                                  [DEPRECATED:--version-number]
+  --version INTEGER RANGE         Number of artifact version.
```
Comment (andersonpablosilva) [inline docs/mlgit_commands.md:197]: Check previous comment about INTEGER RANGE.

File: tests/integration/test_06_commit_files.py [hunk 3]
```diff
@@ -79,9 +79,7 @@ def test_05_commit_command_with_deprecated_version_number(self):
 
         result = check_output(MLGIT_COMMIT % ('dataset', 'dataset' + '-ex', '--version-number=2'))
 
-        self.assertIn(messages[106] % ('--version-number', '--version'), result)
-        self.assertIn(messages[17] % (os.path.join(self.tmp_dir, ML_GIT_DIR, 'dataset', 'metadata'),
-                                      os.path.join('computer-vision', 'images', 'dataset' + '-ex')), result)
+        self.assertIn(messages[106] % '--version-number', result)
```
Comment (andersonpablosilva) [inline tests/integration/test_06_commit_files.py:82]: Maybe we should use the new output_messages list. We should avoid changing or adding messages to the old indexed messages list.

Note: this comment also applies to the following places: 
- tests/integration/output_messages.py: L105, L115
- tests/integration/test_19_create.py: L186

**Review by andersonpablosilva — CHANGES_REQUESTED** [review]

File: docs/mlgit_commands.md (hunk 1 above)
Comment (lucasncabral) [reply docs/mlgit_commands.md:157]: As we use the tag to generate a reference file for ml-git, a tag with a very long version ends up causing errors when trying to generate this file.

In an initial version we worked without defining the range, but it was corrected in PR #25 to deal with the range of a number of up to 9 places.

File: docs/mlgit_commands.md (hunk 2 above)
Comment (lucasncabral) [reply docs/mlgit_commands.md:197]: Answered in the previous comment, please take a look.

**Review by Mun1z — APPROVED** [review]

**Review by GustavoMaciel — APPROVED** [review]

File: docs/mlgit_commands.md (hunk 1 above)
Comment (andersonpablosilva) [reply docs/mlgit_commands.md:157]: Ok, I got it. In this case we can improve the help description and make it clear that this is a range of allowed values.

File: tests/integration/test_06_commit_files.py (hunk 3 above)
Comment (lucasncabral) [reply tests/integration/test_06_commit_files.py]: This and the other messages were sent to the new output_messages list in 89ae397.

File: docs/mlgit_commands.md (hunk 1 above)
Comment (lucasncabral) [reply docs/mlgit_commands.md]: The help message for this option was modified on 89ae397. New message: 

> Set the version number of the artifact. This number must be in the range 0 to 999999999.

**Review by gilMars — APPROVED** [review]

**Review by andersonpablosilva — APPROVED** [review]

**Review by diego-pedroso — APPROVED** [review]

**Pull request merged**